### PR TITLE
WB-1645: Allow custom option item elements

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:

1. Allow custom `OptionItem` components by using `DetailCell` internally.

- Removed `ClickableBehavior` from `OptionItem` and replaced it with
  `DetailCell` (which internally uses `Clickable`).
- Added stories and included docs for `OptionItem`.

2. Modified Cell to support a new prop required for the `Optionitem` changes:
- `aria-selected` is used to allow the `OptionItem` to be selectable (via aria attributes).

Issue: WB-1645

### SingleSelect:

1. Navigate to http://localhost:6061/?path=/docs/dropdown-singleselect--docs#custom-option-items
2. Verify that the custom option items are rendered as expected.
3. Verify that the custom option items are clickable and that clicking on them
   selects them.

https://github.com/Khan/wonder-blocks/assets/843075/6d171969-cc1e-4922-bbcf-16e23cb8e4d4

### MultiSelect:

1. Navigate to http://localhost:6061/?path=/docs/dropdown-multiselect--docs#custom-option-items
2. Verify that the custom option items are rendered as expected.
3. Verify that the custom option items are clickable and that clicking on them
   selects them.

https://github.com/Khan/wonder-blocks/assets/843075/e60716a7-533c-45c4-b239-c282de7dbaf9

### OptionItem docs:

1. Navigate to http://localhost:6061/?path=/docs/dropdown-optionitem--docs
2. Verify that the documentation for `OptionItem` is correct and up to date.
Screenshot 2023-12-13 at 12 07 11 PM

### CODE REVIEW

File: __docs__/wonder-blocks-dropdown/option-item.argtypes.tsx
Comment (jandrade) [inline __docs__/wonder-blocks-dropdown/option-item.argtypes.tsx:1]: **note:** Noticed that we didn't have docs for this, and now that this component is more complex, we should have docs included!

File: packages/wonder-blocks-cell/src/components/internal/cell-core.tsx [hunk 1]
```diff
@@ -172,6 +172,7 @@ const CellCore = (props: CellCoreProps): React.ReactElement => {
         href,
         onClick,
         "aria-label": ariaLabel,
+        "aria-selected": ariaSelected,
```
Comment (jandrade) [inline packages/wonder-blocks-cell/src/components/internal/cell-core.tsx:175]: **note:** This is needed by `OptionItem` so we can know when a given option is selected.

File: packages/wonder-blocks-dropdown/src/components/check.tsx [hunk 2]
```diff
@@ -16,30 +14,17 @@ type CheckProps = {
...
         <PhosphorIcon
             icon={checkIcon}
             size="small"
-            color={
-                disabled
-                    ? offBlack32
-                    : pressed || hovered || focused
-                    ? white
-                    : offBlack
-            }
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/check.tsx:42]: **note:** Removed this as we no longer depend on `Clickable/getClickableBehavior` and instead we assign this with CSS (defined in `OptionItem`).

File: packages/wonder-blocks-dropdown/src/components/checkbox.tsx [hunk 3]
```diff
@@ -18,50 +18,28 @@ type CheckProps = {
...
-        ? blue
-        : white;
-    const checkColor = disabled
-        ? offBlack32
-        : clickInteraction
-        ? pressed
-            ? activeBlue
-            : blue
-        : white;
+    const {disabled, selected} = props;
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/checkbox.tsx:28]: **note:** Similar to `check.tsx`, these states are now handled directly in `OptionItem` with css objects.

File: packages/wonder-blocks-dropdown/src/components/option-item.tsx [hunk 4]
```diff
@@ -124,93 +170,143 @@ export default class OptionItem extends React.Component<OptionProps> {
...
-                    );
-                }}
-            </ClickableBehavior>
+                        )}
+                    </>
+                }
+                rightAccessory={rightAccessory}
+                subtitle1={
+                    <LabelSmall className="subtitle">{subtitle1}</LabelSmall>
+                }
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/option-item.tsx:212]: **note:** Adding this custom "LabelSmall.subtitle" wrapper to be able to style the hover state correctly.

**Review by jeresig — APPROVED** [review]
Very nice - looks great to me!

**Pull request merged**